src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const cloneOfState = { ...state };
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      Object.keys(cloneOfState).forEach(key => delete cloneOfState[key]);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+        if (cloneOfState.hasOwnProperty(actions[i].keysToRemove[y])) {
+          delete cloneOfState[actions[i].keysToRemove[y]];
+        }
+      }
```

Comment:
use `for in` loop  and remove `if` 

If the key doesn't exist in the object and you are trying to remove this key, JS will skip it without any error
![image](https://user-images.githubusercontent.com/37452763/165363987-cc9b4bc1-3b22-4f02-8991-fcd1d66376e7.png)